lib/rust/parser/src/lib.rs

```diff
@@ -262,4 +225,44 @@ mod benches {
             parser.run(&str);
         });
     }
+
+    #[bench]
+    fn bench_blocks(bencher: &mut Bencher) {
+        use rand::prelude::*;
+        use rand_chacha::ChaCha8Rng;
+        let lines = 10_000;
+        let mut str = String::new();
+        let mut rng = ChaCha8Rng::seed_from_u64(0);
+        let mut indent = 0u32;
+        for _ in 0..lines {
+            // Indent:
+            // 1/8 chance of increasing.
+            // 1/8 chance of decreasing.
+            // 3/4 chance of leaving unchanged.
+            match rng.gen_range(0..8) {
+                0u32 => indent = indent.saturating_sub(1),
+                1 => indent += 1,
+                _ => (),
```

Comment:
Are we sure that we want to use random input in benchmarsk? This way it would be hard to compare benches between different commits.

Reply:
`ChaCha8` is deterministic; it produces different output streams for different seeds, but I've initialized it with a fixed seed (0).